File: Baekjoon/DynamicProgramming/[24417]알고리즘 수업 - 피보나치 수 2/solution.cpp
#include <iostream>
#include <vector>

using namespace std;

#define endl "\n"
#define MOD 1000000007

int tmp1, tmp2, res;
int N;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    cin >> N;

    tmp1 = tmp2 = 1;

    for (int i = 3; i <= N; i++)
    {
        res = (tmp1 + tmp2) % MOD;
        tmp1 = tmp2;
        tmp2 = res;
    }

    cout << res << " " << N - 2 << endl;

    return 0;
}

File: Baekjoon/DynamicProgramming/[24417]알고리즘 수업 - 피보나치 수 2/solution.py
N = int(input())

tmp1 = tmp2 = 1
res = 0
MOD = 1000000007

for i in range(3, N + 1):
    res = (tmp1 + tmp2) % MOD
    tmp1 = tmp2
    tmp2 = res

print(res, N - 2)
